_posts/2023-04-19-#27039.md

```diff
@@ -7,20 +7,47 @@ authors: [pinheadmz]
 components: ["block storage"]
 host: stickies-v
 status: upcoming
-commit:
+commit: 93c70287a
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+- Recommended reading from earlier review clubs:
+  - [17994: Flush undo files after last block write](/17994) to learn more about how blocks are downloaded and stored
+  - [16981: Improve runtime performance of --reindex](/16981) and [24858: incorrect blk file size calculation during reindex results in recoverable blk file corruption](/24858) to learn more about reindexing
+
+- The `blocks/` directory contains three datasets:
+  - the `blknnnnn.dat` and `revnnnnn.dat` files which are raw append-only files containing the serialized block data and [undo data](https://bitcoin.stackexchange.com/questions/117887/why-undo-files-are-needed-in-the-bitcoin-core), respectively
+  - the `blocks/index/` block index data which is stored in a LevelDB database. The index helps to quickly find blocks without having to re-scan all the data files.
+    - At runtime, the block index is loaded into [`BlockManager::m_block_index`](https://github.com/bitcoin/bitcoin/blob/19764dc143281376ea08e954018479ed10405b72/src/node/blockstorage.h#L152). However, we also maintain statistics about the block files in [`BlockManager::m_blockfile_info`](https://github.com/bitcoin/bitcoin/blob/19764dc143281376ea08e954018479ed10405b72/src/node/blockstorage.h#L117).
+
+- Flushing is the process of bringing the data that we keep in-memory in sync with the data we store on disk. We want to flush regularly enough to ensure that we can fully recover when bitcoind is unexpectedly killed, but not too regularly that the (generally quite slow) disk I/O slows down performance too much.
+
+- A reindex, amongst other things, recreates the block index from the block files on disk. The block files themselves don't need to be updated: blocks are already assumed to not be stored sequentially in or across block files anyway.
+
+- This PR skips flushing to disk those blocks that already have a known position in the datastore. Skipping this means that users can write-protect the blk files on disk which may be useful for security or even safely sharing that data between multiple bitcoind instances.
+
+- Some of the functions that are relevant to the reindexing process behave differently than can perhaps be expected from their name:
+  - `SaveBlockToDisk()` does not actually write anything to disk when (as is the case for a reindex) the known block position is passed (as `dbp`), but we still need to call it to update the blockfile info stats
```

Comment:
A description of what it *does* do would be helpful, though not 100% necessary. Links to the functions would be nice though!

Reply:
I feel like that's covered in the last part? 

> but we still need to call it to update the blockfile info stats